component/azstorage/file_share.go

```diff
@@ -508,28 +569,162 @@ func (fs *FileShare) List(prefix string, marker *string, count int32) ([]*intern
 	return fileList, listFile.NextMarker.Val, nil
 }
 
+// ReadToFile : Download an Azure file to a local file
 func (fs *FileShare) ReadToFile(name string, offset int64, count int64, fi *os.File) error {
-	return syscall.ENOTSUP
+	log.Trace("FileShare::ReadToFile : name %s, offset : %d, count %d", name, offset, count)
+	//defer exectime.StatTimeCurrentBlock("FileShare::ReadToFile")()
+
+	if offset != 0 {
+		log.Err("FileShare::ReadToFile : offset is not 0")
+		return errors.New("offset is not 0")
+	}
+
+	fileName, dirPath := getFileAndDirFromPath(filepath.Join(fs.Config.prefixPath, name))
+	fileURL := fs.Share.NewDirectoryURL(dirPath).NewFileURL(fileName)
+
+	defer log.TimeTrack(time.Now(), "FileShare::ReadToFile", name)
+	_, err := azfile.DownloadAzureFileToFile(context.Background(), fileURL, fi, fs.downloadOptions)
+
+	if err != nil {
+		e := storeFileErrToErr(err)
+		if e == ErrFileNotFound {
+			return syscall.ENOENT
+		} else {
+			log.Err("FileShare::ReadToFile : Failed to download file %s (%s)", name, err.Error())
+			return err
+		}
+	}
+
+	return nil
 }
 
+// ReadBuffer : Downloads a file to a buffer
 func (fs *FileShare) ReadBuffer(name string, offset int64, len int64) ([]byte, error) {
-	return nil, syscall.ENOTSUP
+	log.Trace("FileShare::ReadBuffer : name %s", name)
+	var buff []byte
+
+	if offset != 0 {
```

Comment:
ReadBuffer method is suppose to read based on offset and length so checking for offset != 0 might not be correct here.

Reply:
We are checking since the SDK doesnt support setting an offset.

Reply:
that is true even in case of block blob as well, either we upload entire file or it overwrites from offset 0, just check what we have done in there and let's keep it symmetric to that.

Reply:
Blob takes in an offset so we pass it